src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -1080,6 +1080,9 @@ void ciEnv::register_method(ciMethod* target,
       MethodData* mdo = method()->method_data();
       if (mdo != nullptr && _inc_decompile_count_on_failure) {
         mdo->inc_decompile_count();
+        if (nm != nullptr && nm->preloaded()) {
```

Comment:
`nm` is not set yet here. You need to check `preload`

Reply:
Right. Actually, I think I need to invalidate the A4 SCC entry anyway, so I can just do that on trapping path, and avoid extending any MDOs. This also avoids any attempt to actually load A4 code, touching any queues, or interfering with compilation policies. I pushed a new version, going to measure its perf tomorrow.